### ISSUE TITLE
Error: spawn .bin\git.CMD ENOENT

### ISSUE THREAD

**Issue by dcbroad3** [bower/bower#2251]
As of a week ago, we started to see the below error from Bower in our build process when it installs packages. The error seems to occur regardless of the version of Bower, but does only happen when using NoGit's git.cmd (https://github.com/whyleee/nogit), not when using git.exe (note the two entries from where git below). This may be an issue with nogit, and if so I'll post there, but I can't tell. I do see that Bower has special handling for non-exe git, for #626, and my best guess is this is related to that issue. It does look like it's trying to spawn the cmd, which is wrong based on https://github.com/nodejs/node-v0.x-archive/issues/2318. I note that the extension ends up in all caps in the error, which seems wrong also, but I'm not sure what to make of it. 

```
D:\{path}>where git
D:\{path}\.bin\git.cmd
C:\Program Files\Git\cmd\git.exe

D:\{path}>bower install
{ not-cached, resolve actions for other packages }
bower malihu-custom-scrollbar-plugin#3.1.3       not-cached https://github.com/malihu/malihu-custom-scrollbar-plugin.git#3.1.3
bower malihu-custom-scrollbar-plugin#3.1.3          resolve https://github.com/malihu/malihu-custom-scrollbar-plugin.git#3.1.3
{ not-cached, resolve, checkout actions for other packages }
bower malihu-custom-scrollbar-plugin#3.1.3         checkout 3.1.3
{ checkout actions for other packages }
bower malihu-custom-scrollbar-plugin#3.1.3           ENOENT spawn .bin\git.CMD ENOENT

Stack trace:
Error: spawn .bin\git.CMD ENOENT
    at exports._errnoException (util.js:855:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:178:32)
    at onErrorNT (internal/child_process.js:344:16)
    at nextTickCallbackWith2Args (node.js:455:9)
    at process._tickCallback (node.js:369:17)

Console trace:
Error
    at StandardRenderer.error (C:\Users\{user}\AppData\Roaming\npm\node_modules\bower\lib\renderers\StandardRenderer.js:81:37)
    at Logger.<anonymous> (C:\Users\{user}\AppData\Roaming\npm\node_modules\bower\lib\bin\bower.js:110:26)
    at emitOne (events.js:77:13)
    at Logger.emit (events.js:169:7)
    at Logger.emit (C:\Users\{user}\AppData\Roaming\npm\node_modules\bower\lib\node_modules\bower-logger\lib\Logger.js:29:39)
    at C:\Users\{user}\AppData\Roaming\npm\node_modules\bower\lib\commands\index.js:48:20
    at _rejected (C:\Users\{user}\AppData\Roaming\npm\node_modules\bower\lib\node_modules\q\q.js:844:24)
    at C:\Users\{user}\AppData\Roaming\npm\node_modules\bower\lib\node_modules\q\q.js:870:30
    at Promise.when (C:\Users\{user}\AppData\Roaming\npm\node_modules\bower\lib\node_modules\q\q.js:1122:31)
    at Promise.promise.promiseDispatch (C:\Users\{user}\AppData\Roaming\npm\node_modules\bower\lib\node_modules\q\q.js:788:41)
System info:
Bower version: 1.7.9
Node version: 5.3.0
OS: Windows_NT 10.0.10586 x64

```

**Comment by dcbroad3** [comment]
Verbose stack trace:

```
Stack trace:
Error: spawn .bin\git.CMD ENOENT
    at exports._errnoException (util.js:855:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:178:32)
    at onErrorNT (internal/child_process.js:344:16)
From previous event:
    at executeCmd (C:\Users\Dbroad\AppData\Roaming\npm\node_modules\bower\lib\util\cmd.js:61:22)
    at PThrottler._processEntry (C:\Users\Dbroad\AppData\Roaming\npm\node_modules\bower\lib\node_modules\p-throttler\index.js:89:25)
    at PThrottler._processQueue (C:\Users\Dbroad\AppData\Roaming\npm\node_modules\bower\lib\node_modules\p-throttler\index.js:67:18)
    at Array.forEach (native)
    at PThrottler._onFulfill (C:\Users\Dbroad\AppData\Roaming\npm\node_modules\bower\lib\node_modules\p-throttler\index.js:121:17)
```

**Comment by dcbroad3** [comment]
Found the cause of my issues, although it still seems there may be something Bower can do to make this work better. 
I had specified the local git path for https://github.com/kmees/MSBuild.NodeTools as ".bin" because that was the location relative to where I was running Bower. Bower initially finds git and continues on. However, after checking out the projects I want to install and going to resolve them, it apparently tries to run git from another location, and can't find the ".bin" folder, then the ENOENT error is thrown because it can't find git. 
If you ask me, this still somewhat indicates an issue, because if Bower is initially able to find git, it shouldn't fail to find it later on through its install process. My fix has been to specify the full path, which is something I'll do either way, but it seems to me that if Bower finds git initially, it shouldn't fail to find it halfway through its install packages process.
